packages/run-protocol/src/vaultFactory/vaultFactory.js

```diff
@@ -75,27 +76,40 @@ export const start = async (zcf, privateArgs) => {
 
   assertElectorateMatches(electorateParamManager, governedTerms);
 
+  /** For temporary staging of newly minted tokens */
+  const { zcfSeat: stage } = zcf.makeEmptySeatKit();
   const { zcfSeat: rewardPoolSeat } = zcf.makeEmptySeatKit();
 
   /**
    * We provide an easy way for the vaultManager to add rewards to
    * the rewardPoolSeat, without directly exposing the rewardPoolSeat to them.
    *
-   * @type {ReallocateWithFee}
+   * @type {MintAndReallocate}
    */
-  const reallocateWithFee = (fee, fromSeat, otherSeat = undefined) => {
-    rewardPoolSeat.incrementBy(
-      fromSeat.decrementBy(
-        harden({
-          RUN: fee,
-        }),
-      ),
-    );
-    if (otherSeat !== undefined) {
-      zcf.reallocate(rewardPoolSeat, fromSeat, otherSeat);
-    } else {
-      zcf.reallocate(rewardPoolSeat, fromSeat);
+  const mintAndReallocate = (toMint, fee, seat, ...otherSeats) => {
+    const wanted = AmountMath.subtract(toMint, fee);
+    runMint.mintGains(harden({ RUN: toMint }), stage);
+    try {
+      rewardPoolSeat.incrementBy(stage.decrementBy(harden({ RUN: fee })));
+      seat.incrementBy(stage.decrementBy(harden({ RUN: wanted })));
+      zcf.reallocate(rewardPoolSeat, stage, seat, ...otherSeats);
+    } catch (e) {
+      stage.clear();
+      rewardPoolSeat.clear();
+      runMint.burnLosses(harden({ RUN: toMint }), stage);
+      throw e;
+    } finally {
+      assert(
+        AmountMath.isEmpty(stage.getAmountAllocated('RUN', runBrand)),
+        X`Stage should be empty of RUN`,
```

Comment:
since it exists to hold newly minted RUN, shouldn't it be empty entirely at this point?

Reply:
Yes probably. I'll seed to figure out a straightforward way to ask that.